File: LC_JS-SOLUTIONS/1984. Minimum Difference Between Highest and Lowest of K Scores.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var minimumDifference = function (nums, k) {
  nums.sort((a, b) => a - b);
  let l = 0;
  let r = k - 1;
  res = Infinity;

  while (r < nums.length) {
    res = Math.min(res, nums[r] - nums[l]);
    r = r + 1;
    l = l + 1;
  }

  return res;
};

File: LC-Python-Solution/1984. Minimum Difference Between Highest and Lowest of K Scores.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:

        nums.sort()
        res = float("inf")
        l = 0 
        r = k-1
        
        while r<len(nums):
            res = min(res, nums[r] - nums[l] )
            l= l + 1
            r= r + 1
        
        return res






        